crates/amaru-ledger/src/rules/transaction/vkey_witness.rs

```diff
@@ -0,0 +1,204 @@
+use std::{array::TryFromSliceError, ops::Deref};
+
+use crate::rules::{context::UtxoSlice, TransactionRuleViolation};
+use amaru_kernel::{
+    Certificate, HasAddress, HasKeyHash, Hash, Hasher, KeepRaw, MintedTransactionBody, NonEmptySet,
+    OriginalHash, PublicKey, Signature, VKeyWitness, Voter,
+};
+
+pub fn validate_vkey_wintesses(
+    transaction_body: &KeepRaw<'_, MintedTransactionBody<'_>>,
+    vkey_witnesses: &Option<NonEmptySet<VKeyWitness>>,
+    utxo_slice: &UtxoSlice,
+) -> Result<(), TransactionRuleViolation> {
+    let empty_vec = vec![];
```

Comment:
All those `empty_vec` look weird. Isn't `unwrap_or_default` useful here?

Reply:
Yeah, I really don't like it. I can't use `unwrap_or_default` though because there is no default implemented for `&Vec<T>`